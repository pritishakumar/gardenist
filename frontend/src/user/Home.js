import React from 'react';
import { Link } from 'react-router-dom';
import { useSelector } from 'react-redux';
// Components within this project
import Register from './Register';
import Login from './Login';
// Material UI Components
import Paper from '@material-ui/core/Paper';
import Grid from '@material-ui/core/Grid';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
// Material UI Style Hook
import { makeStyles } from '@material-ui/core/styles';
const useStyles = makeStyles((theme) => ({
	welcome: {
		padding: '10px;',
	},
	link: {
		textDecoration: 'none',
	},
	paper: {
		margin: "10px",
	},
}));

/** Home/landing page where use sees a welcome 
 * message and offers to either start search, make
 * a new user account or sign into their existing
 * account. Renders ./Login and ./Register Components
 * 
 * Authorization: none, accessible for all
 */
function Home() {
	const user = useSelector((state) => state.user);
	const classes = useStyles();

	return (
		<div>
			<br />
			<Grid container spacing={4} direction="row" justify="space-around" alignItems="center">
				<Grid item align="center" item xs={10} p={2}>
					<Paper className={classes.welcome}>
						<Typography variant="h3" color="inherit">
							Welcome!
						</Typography>
						<p>Go ahead and start researching for your dream garden!</p>
						<Link className={classes.link} to="/search">
							<Button type="submit" variant="contained" color="primary">
								Start Searching!
							</Button>
						</Link>
					</Paper>
				</Grid>

				{!user || !user.email 
        ? (<div>
						<Grid item xs={12}>
							<Paper className={classes.paper}>
								<Register />
							</Paper>
						</Grid>

						<Grid item xs={12}>
							<Paper className={classes.paper}>
								<Login />
							</Paper>
						</Grid>
					</div>) 
        : (null)
        }
			</Grid>
		</div>
	);
}

export default Home;
