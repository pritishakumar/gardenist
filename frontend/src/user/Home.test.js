import React from "react";
import { MemoryRouter } from 'react-router-dom';
import { render, fireEvent } from "@testing-library/react";
import Home from "./Home";
import configureStore from 'redux-mock-store';
import { Provider, connect } from 'react-redux';

// smoke test
it("renders without crashing", function() {
  const mockStore = configureStore();
  const initialState = {};
  const store = mockStore(initialState);

  render(
    <Provider store={store}>
      <MemoryRouter initialEntries = {['/']}>
        <Home />
      </MemoryRouter>
    </Provider>
  );
});

// snapshot test
it("matches snapshot", function() {
  const mockStore = configureStore();
  const initialState = {};
  const store = mockStore(initialState);

  const { asFragment } =  render(
    <Provider store={store}>
      <MemoryRouter initialEntries = {['/']}>
        <Home />
      </MemoryRouter>
    </Provider>
  );
  expect(asFragment()).toMatchSnapshot();
})