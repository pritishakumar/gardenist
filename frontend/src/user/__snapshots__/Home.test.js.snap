// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <br />
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-around"
    >
      <div
        align="center"
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10"
        p="2"
      >
        <div
          class="MuiPaper-root makeStyles-welcome-5 MuiPaper-elevation1 MuiPaper-rounded"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-colorInherit"
          >
            Welcome!
          </h3>
          <p>
            Go ahead and start researching for your dream garden!
          </p>
          <a
            class="makeStyles-link-6"
            href="/search"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                Start Searching!
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </a>
        </div>
      </div>
      <div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiPaper-root makeStyles-paper-7 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiPaper-root MuiCard-root Register MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                align="center"
                class="MuiCardContent-root"
              >
                <h3
                  class="MuiTypography-root MuiTypography-h3"
                >
                  Register for an Account!
                </h3>
                <div
                  class="UserForm"
                >
                  <form
                    action="#"
                  >
                    <div
                      class="MuiFormControl-root"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                        data-shrink="false"
                        for="email"
                      >
                        Email Address: 
                      </label>
                      <div
                        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-input-8 MuiInputBase-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiFilledInput-input"
                          name="email"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="MuiFormControl-root"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                        data-shrink="false"
                        for="name"
                      >
                        First Name: 
                      </label>
                      <div
                        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-input-8 MuiInputBase-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiFilledInput-input"
                          name="name"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="MuiFormControl-root"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                        data-shrink="false"
                        for="password"
                      >
                        Password: 
                      </label>
                      <div
                        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-input-8 MuiInputBase-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiFilledInput-input"
                          name="password"
                          type="password"
                          value=""
                        />
                      </div>
                    </div>
                    <br />
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-input-8 MuiButton-containedPrimary"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Register
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            class="MuiPaper-root makeStyles-paper-7 MuiPaper-elevation1 MuiPaper-rounded"
          >
            <div
              class="MuiPaper-root MuiCard-root Login MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                align="center"
                class="MuiCardContent-root"
              >
                <h3
                  class="MuiTypography-root MuiTypography-h3"
                >
                  Log In As An Existing User
                </h3>
                <div
                  class="UserForm"
                >
                  <form
                    action="#"
                  >
                    <div
                      class="MuiFormControl-root"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                        data-shrink="false"
                        for="email"
                      >
                        Email Address: 
                      </label>
                      <div
                        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-input-8 MuiInputBase-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiFilledInput-input"
                          name="email"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                    <div
                      class="MuiFormControl-root"
                    >
                      <label
                        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                        data-shrink="false"
                        for="password"
                      >
                        Password: 
                      </label>
                      <div
                        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline makeStyles-input-8 MuiInputBase-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiFilledInput-input"
                          name="password"
                          type="password"
                          value=""
                        />
                      </div>
                    </div>
                    <br />
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-input-8 MuiButton-containedPrimary"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Login
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
